lib/base/tlsstream.hpp

```diff
@@ -102,15 +102,25 @@ class UnbufferedAsioTlsStream : public AsioTcpTlsStream
 	void BeforeHandshake(handshake_type type);
 };
 
-class AsioTlsStream : public boost::asio::buffered_stream<UnbufferedAsioTlsStream>
+class AsioTlsStream : public SharedObject, public boost::asio::buffered_stream<UnbufferedAsioTlsStream>
 {
 public:
+	DECLARE_PTR_TYPEDEFS(AsioTlsStream);
+
 	inline
 	AsioTlsStream(boost::asio::io_context& ioContext, boost::asio::ssl::context& sslContext, const String& hostname = String())
 		: AsioTlsStream(UnbufferedAsioTlsStreamParams{ioContext, sslContext, hostname})
 	{
 	}
 
+	static AsioTlsStream::Ptr Make(boost::asio::io_context& ioContext, boost::asio::ssl::context& sslContext, const String& hostname = String())
+	{
+		return new AsioTlsStream(ioContext, sslContext, hostname);
+	}
+
+	void ForceDisconnect();
+	void GracefulDisconnect(boost::asio::io_context::strand strand, boost::asio::yield_context yc);
```

Comment:
```suggestion
	void GracefulDisconnect(boost::asio::io_context::strand& strand, boost::asio::yield_context yc);
```

Reply:
The copy constructor of `strand` say this :):
<img width="527" alt="Bildschirmfoto 2024-11-15 um 13 15 32" src="https://github.com/user-attachments/assets/12725efa-26bf-43ba-9805-e2eae82e7b97">

Meaning, it's just like a shared pointer. So, your suggestion is not wrong, but neither is the current implementation.

Reply:
Actually, using a reference here wouldn't work because that method spawns a coroutine, and there's nothing that keeps the `m_IoStrand` object from the `Http` or `Rpc` class alive till that coroutine finishes.